src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -472,7 +476,28 @@ def get(self, request: Request, project: Project) -> Response:
 
             session = requests.Session()
 
-            apps = appstore_connect.get_apps(session, credentials)
+            try:
+                apps = appstore_connect.get_apps(session, credentials)
+            except appstore_connect.UnauthorizedError:
+                asc_credentials = {
+                    "status": "invalid",
+                    "code": AppConnectAuthenticationError().code,
+                }
+            except appstore_connect.ForbiddenError:
+                asc_credentials = {"status": "invalid", "code": AppConnectForbiddenError().code}
+            else:
+                if apps:
+                    asc_credentials = {"status": "valid"}
+                else:
```

Comment:
Hum, Is this really true?  I think you're now returning invalid even when the credentials were fine but there were not apps to list, which isn't really true is it?  I think you already handle invalid above so lines 489-495 should all be considered valid should they not?

(i strongly suspect the appstore_connect.get_apps() docstring and signature is outdated and that it's `except ValueError` clause is never triggered anymore)

Reply:
> Hum, Is this really true? 

i don't think it's necessarily true, but it is what the code has historically been doing when apps aren't available. i'm open to the idea of changing this to `Forbidden` as it's likely that this is will happen when the token hasn't been granted permissions to an app (or there are no apps at all..?), but it's not too clear to me what you're hoping to see at this line instead.

honestly, i don't really have the right tools and permissions to be setting up these test scenarios myself as i'm unable to create API tokens, so i'm really just working off of fumes here.

whatever change is made here, i'm pretty sure it'll be new behaviour relative to the code prior to this change. i'm a little hesitant to do that, but if you feel it's important to add it in to this PR i can do that.

Reply:
apologies, I wasn't very clear here.  I'm concerned that this tells users their credentials are invalid when they have no apps.  I don't think we need to fix it now as it's a funny edge case and it probably is simpler to leave like this.  I've created a ticket to track this as future work though.